File: desarrollo/bk/ColabManager360.Aplication/Services/Helper/HelperService.cs
using ColabManager360.Domain.Entities.Activity.Responses;
using ColabManager360.Domain.Entities.Client;
using ColabManager360.Domain.Entities.Common;
using ColabManager360.Domain.Entities.Common.Requests;
using ColabManager360.Domain.Entities.Helper.Responses;
using ColabManager360.Domain.Interfaces.Helper;

namespace ColabManager360.Aplication.Services.Helper
{
    public class HelperService : IHelperService
    {
        private readonly IHelperRepository _HelperRepository;

        public HelperService(IHelperRepository helperRepository)
        {
            _HelperRepository = helperRepository;
        }
        public async Task<AccountListResponse> AccountLists()
        {

            var response = await _HelperRepository.AccountLists();

            return response;

        }
        public async Task<List<Ubigeo>> GetDepartments() {
            var response = await _HelperRepository.GetDepartments();

            return response;

        }
        public async Task<List<Ubigeo>> GetProvinces(string DepartmentCode) {
            var response = await _HelperRepository.GetProvinces(DepartmentCode);

            return response;

        }
        public async Task<List<Ubigeo>> GetDistricts(string DepartmentCode, string ProvinceCode) {
            var response = await _HelperRepository.GetDistricts(DepartmentCode, ProvinceCode);

            return response;

        }
        public async Task<List<Country>> GetCountries()
        {
            var response = await _HelperRepository.GetCountries();

            return response;

        }

        public async Task<List<MasterDetailTable>> GetMasterList(MasterDetailTableRequest request)
        {
            var response = await _HelperRepository.GetMasterList(request);

            return response;

        }

        public async Task<List<ServiceType>> GetServiceTypes()
        {
            var response = await _HelperRepository.GetServiceTypes();

            return response;

        }

        public async Task<List<CompanyService>> GetCompanyServicesList(int request)
        {
            var response = await _HelperRepository.GetCompanyServicesList(request);

            return response;

        }

        public async Task<List<ActivityType>> GetCompanyActivityTypeList(int request)
        {
            var response = await _HelperRepository.GetCompanyActivityTypeList(request);

            return response;

        }

        public async Task<List<PeriodWeekResponse>> GetPeriodWeekList()
        {
            var response = await _HelperRepository.GetPeriodWeekList();

            return response;

        }
    }
}

File: desarrollo/prueba/src/app/services/helper.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { AccountListResponse } from '../models/helper/account-list-response.model';
import { Ubigeo } from '../models/common/ubigeo.model';
import { Country } from '../models/common/country.model';
import { MasterDetailTable } from '../models/common/master-detail-table.model';
import { Company } from '../models/admin/company.model';
import { CompanyService } from '../models/admin/company-service.model';
import { ServiceType } from '../models/common/service-type.model';
import { ActivityType } from '../models/common/activity-type.model';
import { PeriodWeekResponse } from '../models/activity/period-week-response.model';


@Injectable({
  providedIn: 'root'
})
export class HelperService {

  private apiUrl = environment.app.apiUrl;
  private header = new HttpHeaders().set('Content-Type', 'application/json')

  constructor(
    private http: HttpClient
  ) { }

  AccountLists(): Observable<AccountListResponse> {

    return this.http.get<AccountListResponse>(this.apiUrl + `Helper/AccountLists`, { headers: this.header, withCredentials: true });

  }

  Provinces(departmentCode: string): Observable<Ubigeo[]> {

    return this.http.get<Ubigeo[]>(this.apiUrl + `Helper/Provinces?DepartmentCode=` + departmentCode, { headers: this.header, withCredentials: true });

  }

  Districts(departmentCode: string, provinceCode: string): Observable<Ubigeo[]> {

    return this.http.get<Ubigeo[]>(this.apiUrl + `Helper/Districts?DepartmentCode=${departmentCode}&ProvinceCode=${provinceCode}`, { headers: this.header, withCredentials: true });

  }

  Countries() {

    return this.http.get<Country[]>(this.apiUrl + `Helper/Countries`, { headers: this.header, withCredentials: true });

  }

  MasterTableList(TableName: string, TableCode: string) {

    return this.http.get<MasterDetailTable[]>(this.apiUrl + `Helper/MasterTable/${TableName}/${TableCode}`, { headers: this.header, withCredentials: true });

  }

  CompanyList() {

    return this.http.get<Company[]>(this.apiUrl + `Helper/Company/List`, { headers: this.header, withCredentials: true });
  }

  ServicesTypesList() {

    return this.http.get<ServiceType[]>(this.apiUrl + `Helper/Services/Types/List`, { headers: this.header, withCredentials: true });
  }

  CompanyServicesList(CompanyId: number) {

    return this.http.get<CompanyService[]>(this.apiUrl + `Helper/Company/Services/List/${CompanyId}`, { headers: this.header, withCredentials: true });
  }

  CompanyActivityTypeList(CompanyId: number) {

    return this.http.get<ActivityType[]>(this.apiUrl + `Helper/Company/ActivityType/List/${CompanyId}`, { headers: this.header, withCredentials: true });
  }

  
  PeriodWeekList() {

    return this.http.get<PeriodWeekResponse[]>(this.apiUrl + `Helper/Period/Week/List`, { headers: this.header, withCredentials: true });
  }


}
